### PR TITLE
Fixed walking animations and pathing bugs.

### PR DESCRIPTION
**Pull request by essial**

### CODE REVIEW

File: d2common/math.go [hunk 1]
```diff
@@ -89,3 +91,36 @@ func GetRadiansBetween(p1X, p1Y, p2X, p2Y float64) float64 {
...
+		return finalValue, 0
+	}
+
+	if finalValue < targetvalue {
+		return targetvalue, RoundToDecial(finalValue-targetvalue, 6)
+	}
+	return finalValue, 0
+}
+
+func RoundToDecial(f float64, d int) float64 {
```
Comment (nicholas-eden) [inline d2common/math.go:123]: 😆

File: d2common/math.go [hunk 2]
```diff
@@ -89,3 +91,36 @@ func GetRadiansBetween(p1X, p1Y, p2X, p2Y float64) float64 {
...
+func AdjustWithRemainder(sourceValue, adjustment, targetvalue float64) (newValue, remainder float64) {
+	if adjustment == 0 || math.Abs(adjustment) < 0.000001 {
+		return sourceValue, 0
+	}
+	adjustNegative := adjustment < 0.0
+	maxNegative := targetvalue-sourceValue < 0.0
+	if adjustNegative != maxNegative {
+		// FIXME: This shouldn't happen but it happens all the time..
+		return sourceValue, 0
+		//panic("Cannot move towards the opposite direction...")
```
Comment (carrelld) [inline d2common/math.go:105]: delete this

File: d2common/math.go [hunk 3]
```diff
@@ -89,3 +91,36 @@ func GetRadiansBetween(p1X, p1Y, p2X, p2Y float64) float64 {
...
+	}
+
+	finalValue := sourceValue + adjustment
+	if !adjustNegative {
+		if finalValue > targetvalue {
+			diff := finalValue - targetvalue //  RoundToDecial(finalValue-targetvalue, 6)
```
Comment (carrelld) [inline d2common/math.go:111]: Is this comment useful?

File: d2core/d2map/animated_composite.go [hunk 4]
```diff
@@ -10,9 +10,11 @@ import (
 // AnimatedComposite represents a composite of animations that can be projected onto the map.
 type AnimatedComposite struct {
 	mapEntity
-	animationMode string
-	composite     *d2asset.Composite
-	direction     int
+	//animationMode string
```
Comment (carrelld) [inline d2core/d2map/animated_composite.go:13]: should this be removed or uncommented?

File: d2core/d2map/map_entity.go [hunk 5]
```diff
@@ -75,40 +75,48 @@ func (m *mapEntity) Step(tickTime float64) {
 	}
 
 	stepX, stepY := m.getStepLength(tickTime)
+	looped := false
```
Comment (carrelld) [inline d2core/d2map/map_entity.go:78]: useless variable, remove

File: d2core/d2map/animated_composite.go [hunk 6]
```diff
@@ -61,19 +69,24 @@ func (ac *AnimatedComposite) Render(target d2render.Surface) {
...
 	// TODO: Check if is in town and if is player.
-	newAnimationMode := ac.animationMode
+	newAnimationMode := ac.composite.GetAnimationMode()
 	if !ac.IsAtTarget() {
-		newAnimationMode = d2enum.AnimationModeMonsterWalk.String()
-	}
-
-	if newAnimationMode != ac.animationMode {
-		ac.SetMode(newAnimationMode, ac.weaponClass, ac.direction)
+		if ac.player != nil {
```
Comment (carrelld) [inline d2core/d2map/animated_composite.go:74]: could this nesting be improved?

File: d2common/math.go (hunk 1 above)
Comment (essial) [reply d2common/math.go:123]: The struggle was real.

File: d2core/d2map/animated_composite.go (hunk 4 above)
Comment (essial) [reply d2core/d2map/animated_composite.go:13]: I have a habit of commenting stuff out then removing it later on, I'll try to get better at that.

File: d2common/math.go (hunk 3 above)
Comment (essial) [reply d2common/math.go:111]: Not anymore, I'll remove it.

File: d2core/d2map/map_entity.go [hunk 7]
```diff
@@ -75,40 +75,48 @@ func (m *mapEntity) Step(tickTime float64) {
...
+	for {
+		looped = looped
```
Comment (nicholas-eden) [inline d2core/d2map/map_entity.go:80]: 🤨

Comment (essial) [reply d2core/d2map/map_entity.go:80]: I'm fixing these and re-pushing, this was for debugging. Good catch!

File: d2common/math.go (hunk 1 above)
Comment (nicholas-eden) [reply d2common/math.go:123]: I know you have GoLand.  Fix the squiggly lines!

**Pull request merged**